packages/turbo-workspaces/src/updateDependencies.ts

```diff
@@ -0,0 +1,127 @@
+import fs from "fs-extra";
+import chalk from "chalk";
+import path from "path";
+import {
+  Project,
+  Workspace,
+  DependencyList,
+  PackageManagerDetails,
+  Options,
+  PackageJsonDependencies,
+} from "./types";
+import { Logger } from "./logger";
+import { getPackageJson } from "./utils";
+
+function updateDependencyList({
+  dependencyList,
+  project,
+  to,
+}: {
+  dependencyList: DependencyList;
+  project: Project;
+  to: PackageManagerDetails;
+}): { dependencyList: DependencyList; updated: Array<string> } {
+  const updated: Array<string> = [];
+  project.workspaceData.workspaces.forEach((workspace) => {
+    const { name } = workspace;
+    if (dependencyList[name]) {
+      const workspaceVersion = dependencyList[name];
+      const version = workspaceVersion.startsWith("workspace:")
+        ? workspaceVersion.slice("workspace:".length)
+        : workspaceVersion;
+      dependencyList[name] =
+        to.name === "pnpm" ? `workspace:${version}` : version;
+      updated.push(name);
+    }
+  });
+
+  return { dependencyList, updated };
+}
+
+export default function updateDependencies({
```

Comment:
🤨 Don't love this method but not a blocker. Feels like it should be per-package-manager, with a default implementation.